misc/userscripts/qute-1pass

```diff
@@ -66,59 +66,147 @@ URL=$(echo "$QUTE_URL" | awk -F/ '{print $3}' | sed 's/www.//g')
+}
+
+clipboard-copy() {
+    if [[ -n "$QUTE_1PASS_CLIPBOARD" ]]; then
+        $QUTE_1PASS_CLIPBOARD
+        return
+    fi
+
+    if [[ -n "$WAYLAND_DISPLAY" ]] && command -v wl-copy >/dev/null; then
+        wl-copy --trim-newline
+        return
+    fi
+
+    xclip -in -selection clipboard
+}
+
+if ! command -v op > /dev/null; then
+    echo "message-error 'Missing required command-line tool: op'" >> "$QUTE_FIFO"
+    exit 0
+fi
+
+if ! command -v jq > /dev/null; then
+    echo "message-error 'Missing required command-line tool: jq'" >> "$QUTE_FIFO"
+    exit 0
+fi
+
+OP_VERSION="$(op --version)"
+OP_MAJOR_VERSION="$(echo "$OP_VERSION" | grep --only-matching '^[0-9]')"
+
+if [[ "$OP_MAJOR_VERSION" -lt 2 ]]; then
+    echo "message-error 'Requires op CLI v2.0.0 or higher, but got: $OP_VERSION'" >> "$QUTE_FIFO"
+    exit 0
+fi
+
 echo "message-info 'Looking for password for $URL...'" >> "$QUTE_FIFO"
 
+JQ_TITLE_EXPR='.title + " (in vault \"" + .vault.name + "\")"'
+
 if [ -f "$TOKEN_CACHE" ]; then
-    TOKEN=$(cat "$TOKEN_CACHE")
-    if ! op signin --session="$TOKEN" --output=raw > /dev/null; then
-        TOKEN=$(rofi -dmenu -password -p "1password: "| op signin --output=raw) || TOKEN=""
+    TOKEN="$(cat "$TOKEN_CACHE")"
+    if ! op signin --raw --session="$TOKEN" > /dev/null; then
+        TOKEN="$(dmenu-prompt-pass "1password: "| op signin --raw)" || TOKEN=""
         echo "$TOKEN" > "$TOKEN_CACHE"
     fi
 else
-    TOKEN=$(rofi -dmenu -password -p "1password: "| op signin --output=raw) || TOKEN=""
+    TOKEN=$(dmenu-prompt-pass "1password: "| op signin --raw) || TOKEN=""
     install -m 600 /dev/null "$TOKEN_CACHE"
     echo "$TOKEN" > "$TOKEN_CACHE"
 fi
 
 
-if [ -n "$TOKEN" ]; then
-    UUID=$(op list items --cache --session="$TOKEN" | jq --arg url "$URL" -r '[.[] | {uuid, url: [.overview.URLs[]?.u, .overview.url][]?} | select(.uuid != null) | select(.url != null) | select(.url|test(".*\($url).*"))][.0].uuid') || UUID="" 
-
-    if [ -z "$UUID" ] || [ "$UUID" == "null" ];then
-        echo "message-error 'No entry found for $URL'" >> "$QUTE_FIFO"
-        TITLE=$(op list items --cache --session="$TOKEN" | jq -r '.[].overview.title' | rofi -dmenu -i) || TITLE=""
-        if [ -n "$TITLE" ]; then
-            UUID=$(op list items --cache --session="$TOKEN" | jq --arg title "$TITLE" -r '[.[] | {uuid, title:.overview.title}|select(.title|test("\($title)"))][.0].uuid') || UUID=""
-        else
-            UUID=""
-        fi
-    fi
-
-    if [ -n "$UUID" ];then
-        ITEM=$(op get item --cache --session="$TOKEN" "$UUID")
+if [[ -z "$TOKEN" ]]; then
+    echo "message-error 'Wrong master password'" >> "$QUTE_FIFO"
+    exit 0
+fi
 
-        PASSWORD=$(echo "$ITEM" | jq -r '.details.fields | .[] | select(.designation=="password") | .value')
+if ! LIST_ITEM_OUT="$(op item list --cache --session="$TOKEN" --format=json)"; then
+    echo "message-error 'Failed to list items. See :process for more info'" >> "$QUTE_FIFO"
+    exit 0
```

Comment:
Afaik if the script exists with 1 the text "See :process...." is added automatically.